src/components/PrometheusGraph/PrometheusGraph.js

```diff
@@ -226,15 +228,21 @@ class PrometheusGraph extends React.PureComponent {
     if (this.props.multiSeries !== nextProps.multiSeries) {
       this.prepareMultiSeries(nextProps);
     }
+    if (this.props.selectedLegendKeys !== nextProps.selectedLegendKeys) {
```

Comment:
Do you need to do a shallow array comparison here?

Reply:
I think we may assume that whatever gets passed as `selectedLegendKeys` here comes originally from either a Redux or a local component state, both of which get their references updated as soon as their content changes.

It's an assumption we're making in (almost) all the places where we're checking if the props have changed really.